src/st/e_breakable_with_debris.h

```diff
@@ -0,0 +1,232 @@
+        case CANDELABRA_TALL:
+        case CANDELABRA_SHORT:
+            self->facingLeft = GetSideToPlayer() & 1;
+#ifdef INVERTED_STAGE
+            posY = self->posY.i.hi + 40;
+#else
+            posY = self->posY.i.hi - 40;
+#endif
+            if (breakableType == CANDELABRA_TALL) {
+                debrisCount = 4;
+            } else {
+                debrisCount = 3;
+            }
+            debrisOffsetsY = candelabra_debris_offsets_y;
+            if (breakableType == CANDELABRA_SHORT) {
+                debrisOffsetsY += 5;
+            }
+            for (debrisIndex = 0; debrisIndex < debrisCount; debrisIndex++) {
+                entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+                if (entity != NULL) {
+                    CreateEntityFromEntity(
+                        E_ID(BREAKABLE_DEBRIS), self, entity);
+                    entity->posY.i.hi = posY;
+                    entity->params = debrisOffsetsY[debrisIndex];
+                    entity->facingLeft = self->facingLeft;
+                }
+                entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+                if (entity != NULL) {
+                    CreateEntityFromEntity(E_EXPLOSION, self, entity);
+                    entity->posY.i.hi = posY;
+                    entity->params = EXPLOSION_SMALL;
+                }
+#ifdef INVERTED_STAGE
+                posY -= 16;
+#else
+                posY += 16;
+#endif
+            }
+            g_api.PlaySfx(CANDELABRA_HIT_SFX);
+            break;
+        case BUST:
+            entity = AllocEntity(&g_Entities[160], &g_Entities[192]);
+            if (entity != NULL) {
+                CreateEntityFromCurrentEntity(E_ID(BREAKABLE_DEBRIS), entity);
+                entity->params = 256;
+            }
+            g_api.PlaySfx(SFX_GLASS_BREAK_E);
+#ifdef PREVENT_BUST_RESPAWN
+            entity = AllocEntity(&g_Entities[160], &g_Entities[192]);
+            if (entity != NULL) {
+                CreateEntityFromEntity(E_HEART_DROP, self, entity);
+                entity->params = self->params & 0x1FF;
+            }
+            PreventEntityFromRespawning(self);
+            DestroyEntity(self);
+            return;
+#endif
+            break;
+        case URN:
+            g_api.PlaySfx(SFX_GLASS_BREAK_E);
+            entity = AllocEntity(&g_Entities[160], &g_Entities[192]);
+            if (entity != NULL) {
+                CreateEntityFromEntity(E_HEART_DROP, self, entity);
+                entity->params = self->params & 0x1FF;
+            }
+            PreventEntityFromRespawning(self);
+            DestroyEntity(self);
+            return;
+        case JUG:
+            g_api.PlaySfx(SFX_GLASS_BREAK_E);
+            entity = AllocEntity(&g_Entities[160], &g_Entities[192]);
+            if (entity != NULL) {
+                CreateEntityFromEntity(E_HEART_DROP, self, entity);
+#ifdef JUG_PARAMS
+                entity->params = JUG_PARAMS;
```

Comment:
What concerns me about this or similar approaches for code sharing is how the alternative code paths are expressed and documented. When I see `JUG_PARAMS`, it doesn’t communicate _why_ `entity->params` is assigned differently, nor does it indicate where that value comes from or what it represents. To understand it, I have to navigate to its definition (if the macro definition is unique) or perform a full search across the codebase, which makes the logic harder to follow.

A clearer approach would be to use something more descriptive, like `#ifdef STAGE_NO1`, and assign the value directly (e.g. `entity->params = 3`). That way, the condition and the effect are both visible at a glance, making it immediately clear which overlay uses the alternate code path and what value is being set.

Reply:
So, if we know the purpose it is better to align the macro with the purpose and if we don't know the purpose, to align it with what we do know instead of abstracting it away?
So in this case, params is the index of the item drop.  We would want to do rename JUG_PARAMS to ITEM_DROP_INDEX?
The logic that I'm following is that JUG_PARAMS forces you to start at the very beginning where STAGE_IS_NO1, while not ideal, at least allows you to start at the point where things become unknown.  Is that accurate?

Reply:
In this case it is preferable for the reasons I explained above. Yes.